data-pipeline/src/trace_exporter.rs

```diff
@@ -197,13 +235,29 @@ impl TraceExporter {
             })
     }
 
+    /// Emit a health metric to dogstatsd
+    fn emit_metric(&self, metric: HealthMetric, custom_tags: Option<Vec<&Tag>>) {
```

Comment:
Not sure if it is worth it performance wise. But using a closure to lazily create the metric and tags can avoid computation and allocation when stats are disabled.

Reply:
How likely are metrics to be disabled? If there is a decent chance we definitely should avoid any extra computation and allocation.

Reply:
I'm not sure I follow how one would do this / where the performance benefit would be? If the client is `None` then this function does nothing and just returns? 🤔 (or do you mean allocating the `HealthMetric` type?)

Reply:
Yes the HealthMetric and Vec still need to be allocated. Another option would be to use a macro to add a `if flusher.is_some() {...}` everywhere we create metrics